### PR TITLE
#20765 use <undefined> name for nullable data types

### LINKED ISSUE #20765: NPE on the result set column tooltip
### Description

During the script execution, I found that the driver does not always return the correct typeName. And a null value is not expected.

```
java.lang.NullPointerException: Cannot invoke "String.toLowerCase(java.util.Locale)" because "typeName" is null
	at org.jkiss.dbeaver.ext.mssql.model.SQLServerDialect.getColumnTypeModifiers(SQLServerDialect.java:201)
	at org.jkiss.dbeaver.model.sql.SQLUtils.getColumnTypeModifiers(SQLUtils.java:615)
	at org.jkiss.dbeaver.model.DBUtils.getFullTypeName(DBUtils.java:1736)
	at org.jkiss.dbeaver.model.data.DBDAttributeBindingMeta.getFullTypeName(DBDAttributeBindingMeta.java:123)
	at org.jkiss.dbeaver.ui.controls.resultset.spreadsheet.SpreadsheetPresentation$GridLabelProvider.getToolTipText(SpreadsheetPresentation.java:2781)
	at org.jkiss.dbeaver.ui.controls.lightgrid.GridColumn.getHeaderTooltip(GridColumn.java:351)
	at org.jkiss.dbeaver.ui.controls.lightgrid.LightGrid.handleCellHover(LightGrid.java:4210)
	at org.jkiss.dbeaver.ui.controls.lightgrid.LightGrid.handleHovering(LightGrid.java:3765)
	at org.jkiss.dbeaver.ui.controls.lightgrid.LightGrid.onMouseMove(LightGrid.java:3604)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:216)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4274)
```

2023-08-02 11_37_41-DBeaver Ultimate 23 1 3 - _demo_ Script-37

Script: 

```sql
DECLARE @SQLString VARCHAR(500)
BEGIN
    SELECT @SQLString = 'select result_time from demo.dbo.synchronize_results' 
END
EXECUTE(@SQLString)
```

### DBeaver Version

22.1.3

### Database and driver

server: SQL Anywhere 12.0.1.3994
driver: jConnect (TM) for JDBC (TM) jConnect (TM) for JDBC(TM)/16.0 SP03 PL02 (Build 27403)

### Steps to reproduce

Execute script
Focus on the column name.

Expected: Column name and status
Actual: NPE and no column name and status

### PR DESCRIPTION
**Pull request by LonwoLonwo**
2023-08-02 11_57_15-DBeaver Ultimate 23 1 3 - _demo_ Script-37

You can suggest another name instead of `<undefined>`

For tester: ask me about the database to check

### CODE REVIEW

**Comment by E1izabeth** [comment]
`<undefined>` tells nothing about what is undefined. I'd prefer I little bit longer, but clearer: unknown type

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBUtils.java [hunk 1]
```diff
@@ -1733,6 +1733,10 @@ public static String getFullTypeName(@NotNull DBSTypedObject typedObject)
     @NotNull
     public static String getFullTypeName(DBPDataSource dataSource, @NotNull DBSTypedObject typedObject) {
         String typeName = typedObject.getTypeName();
+        if (CommonUtils.isEmpty(typeName)) {
+            // No answer from the driver side
+            return "<unknown type>";
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBUtils.java]: Please add it into i18n messages

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**